--- postfix-2.11.5.orig/src/util/Makefile.in.orig	2015-08-11 13:48:16.897867369 +0200
+++ postfix-2.11.5/src/util/Makefile.in	2015-08-11 13:54:19.074562492 +0200
@@ -137,7 +137,7 @@
 	gcc -shared -Wl,-soname,dict_pcre.so -o $@ $? -lpcre -L. -lutil
 
 $(LIB):	$(OBJS)
-	gcc -shared -Wl,-soname,libpostfix-util.so.1 -o $(LIB) $(OBJS) -ldl -lcdb -lnsl -ldb
+	gcc -shared -Wl,-soname,libpostfix-util.so.1 -o $(LIB) $(OBJS) -ldl -lnsl -ldb
 
 $(LIB_DIR)/$(LIB): $(LIB)
 	cp $(LIB) $(LIB_DIR)
