diff -uNr old-postfix-2.0.16/src/global/mail_params.c postfix-2.0.16/src/global/mail_params.c
--- old-postfix-2.0.16/src/global/mail_params.c	Thu Sep 25 17:25:19 2003
+++ postfix-2.0.16/src/global/mail_params.c	Thu Sep 25 17:34:14 2003
@@ -76,6 +76,7 @@
 /*	char	*var_debug_peer_list;
 /*	int	var_debug_peer_level;
 /*	int	var_in_flow_delay;
+/*	char    *var_reject_reply_msg_access_denied;	
 /*	int	var_fault_inj_code;
 /*	char   *var_bounce_service;
 /*	char   *var_cleanup_service;
@@ -255,6 +256,7 @@
 char   *var_export_environ;
 char   *var_debug_peer_list;
 int     var_debug_peer_level;
+char   *var_reject_reply_msg_access_denied;
 int     var_fault_inj_code;
 char   *var_bounce_service;
 char   *var_cleanup_service;
@@ -512,6 +514,7 @@
 	VAR_SMTP_TLS_CLIST, DEF_SMTP_TLS_CLIST, &var_smtp_tls_cipherlist, 0, 0,
 	VAR_SMTP_TLS_SCACHE_DB, DEF_SMTP_TLS_SCACHE_DB, &var_smtp_tls_scache_db, 0, 0,
 	VAR_TLS_DAEMON_RAND_SOURCE, DEF_TLS_DAEMON_RAND_SOURCE, &var_tls_daemon_rand_source, 0, 0,
+	VAR_REJECT_REPLY_MSG_ACCESS_DENIED, DEF_REJECT_REPLY_MSG_ACCESS_DENIED, &var_reject_reply_msg_access_denied, 1, 0,
 	0,
     };
     static CONFIG_STR_FN_TABLE function_str_defaults_2[] = {
diff -uNr old-postfix-2.0.16/src/global/mail_params.h postfix-2.0.16/src/global/mail_params.h
--- old-postfix-2.0.16/src/global/mail_params.h	Thu Sep 25 17:25:19 2003
+++ postfix-2.0.16/src/global/mail_params.h	Thu Sep 25 17:35:10 2003
@@ -1762,6 +1762,13 @@
 #define DEF_RESOLVE_DEQUOTED		1
 extern bool var_resolve_dequoted;
 
+/*
+ * SMTPD messages
+ */
+#define VAR_REJECT_REPLY_MSG_ACCESS_DENIED	"smtpd_reject_msg_access_denied"
+#define DEF_REJECT_REPLY_MSG_ACCESS_DENIED	"rejected: Relay access denied."
+extern char *var_reject_reply_msg_access_denied;
+
  /*
   * Service names. The transport (TCP, FIFO or UNIX-domain) type is frozen
   * because you cannot simply mix them, and accessibility (private/public) is
diff -uNr old-postfix-2.0.16/src/smtpd/smtpd.c postfix-2.0.16/src/smtpd/smtpd.c
--- old-postfix-2.0.16/src/smtpd/smtpd.c	Thu Sep 25 17:25:19 2003
+++ postfix-2.0.16/src/smtpd/smtpd.c	Thu Sep 25 17:37:20 2003
@@ -2011,6 +2011,7 @@
 	VAR_VERP_CLIENTS, DEF_VERP_CLIENTS, &var_verp_clients, 0, 0,
 	VAR_RELAY_CCERTS, DEF_RELAY_CCERTS, &var_relay_ccerts, 0, 0,
 	VAR_SMTPD_SASL_TLS_OPTS, DEF_SMTPD_SASL_TLS_OPTS, &var_smtpd_sasl_tls_opts, 0, 0,
+	VAR_REJECT_REPLY_MSG_ACCESS_DENIED, DEF_REJECT_REPLY_MSG_ACCESS_DENIED, &var_reject_reply_msg_access_denied, 1, 0,
 	0,
     };
     static CONFIG_RAW_TABLE raw_table[] = {
diff -uNr old-postfix-2.0.16/src/smtpd/smtpd_check.c postfix-2.0.16/src/smtpd/smtpd_check.c
--- old-postfix-2.0.16/src/smtpd/smtpd_check.c	Thu Sep 25 17:25:19 2003
+++ postfix-2.0.16/src/smtpd/smtpd_check.c	Thu Sep 25 17:37:05 2003
@@ -1175,6 +1175,7 @@
 			               char *reply_name, char *reply_class)
 {
     char   *myname = "check_relay_domains";
+    	VSTRING *reply_msg = vstring_alloc(1);
 
 #if 1
     static int once;
@@ -1204,9 +1205,10 @@
     /*
      * Deny relaying between sites that both are not in relay_domains.
      */
+    vstring_strcat(reply_msg, "%d <%s>: %s ");
+    vstring_strcat(reply_msg, var_reject_reply_msg_access_denied);
     return (smtpd_check_reject(state, MAIL_ERROR_POLICY,
-			       "%d <%s>: %s rejected: Relay access denied",
-			       var_relay_code, reply_name, reply_class));
+			    vstring_export(reply_msg), var_relay_code, reply_name, reply_class));
 }
 
 /* permit_auth_destination - OK for message relaying */
@@ -1263,6 +1265,7 @@
 static int reject_unauth_destination(SMTPD_STATE *state, char *recipient)
 {
     char   *myname = "reject_unauth_destination";
+    	VSTRING *reply_msg = vstring_alloc(1);
 
     if (msg_verbose)
 	msg_info("%s: %s", myname, recipient);
@@ -1276,9 +1279,10 @@
     /*
      * Reject relaying to sites that are not listed in relay_domains.
      */
+    vstring_strcat(reply_msg, "%d <%s>: ");
+    vstring_strcat(reply_msg, var_reject_reply_msg_access_denied);
     return (smtpd_check_reject(state, MAIL_ERROR_POLICY,
-			       "%d <%s>: Relay access denied",
-			       var_relay_code, recipient));
+			    vstring_export(reply_msg), var_relay_code, recipient));
 }
 
 /* reject_unauth_pipelining - reject improper use of SMTP command pipelining */
