diff -urN snapshot-20011127/src/global/mail_params.c snapshot-20011127.dzikus/src/global/mail_params.c
--- snapshot-20011127/src/global/mail_params.c	Thu Nov 29 12:38:01 2001
+++ snapshot-20011127.dzikus/src/global/mail_params.c	Thu Nov 29 12:36:52 2001
@@ -71,6 +71,7 @@
 /*	char	*var_debug_peer_list;
 /*	int	var_debug_peer_level;
 /*	int	var_in_flow_delay;
+/*	char    *var_reject_reply_msg_access_denied;
 /*	int	var_fault_inj_code;
 /*
 /*	void	mail_params_init()
@@ -220,6 +221,7 @@
 char   *var_export_environ;
 char   *var_debug_peer_list;
 int     var_debug_peer_level;
+char   *var_reject_reply_msg_access_denied;
 int     var_fault_inj_code;
 
 /* check_myhostname - lookup hostname and validate */
@@ -362,6 +364,7 @@
 	VAR_SMTP_TLS_CLIST, DEF_SMTP_TLS_CLIST, &var_smtp_tls_cipherlist, 0, 0,
 	VAR_SMTP_TLS_SCACHE_DB, DEF_SMTP_TLS_SCACHE_DB, &var_smtp_tls_scache_db, 0, 0,
 	VAR_TLS_DAEMON_RAND_SOURCE, DEF_TLS_DAEMON_RAND_SOURCE, &var_tls_daemon_rand_source, 0, 0,
+	VAR_REJECT_REPLY_MSG_ACCESS_DENIED, DEF_REJECT_REPLY_MSG_ACCESS_DENIED, &var_reject_reply_msg_access_denied, 1, 0, 
 	0,
     };
     static CONFIG_STR_FN_TABLE function_str_defaults_2[] = {
diff -urN snapshot-20011127/src/global/mail_params.h snapshot-20011127.dzikus/src/global/mail_params.h
--- snapshot-20011127/src/global/mail_params.h	Thu Nov 29 12:38:01 2001
+++ snapshot-20011127.dzikus/src/global/mail_params.h	Thu Nov 29 12:32:18 2001
@@ -1539,6 +1539,13 @@
 #define DEF_FAULT_INJ_CODE		0
 extern int var_fault_inj_code;
 
+ /*
+  * SMTPD messages
+  */
+#define VAR_REJECT_REPLY_MSG_ACCESS_DENIED	"smtpd_reject_msg_access_denied"
+#define DEF_REJECT_REPLY_MSG_ACCESS_DENIED	"rejected: Relay access denied."
+extern char *var_reject_reply_msg_access_denied;
+
 /* LICENSE
 /* .ad
 /* .fi
diff -urN snapshot-20011127/src/smtpd/smtpd.c snapshot-20011127.dzikus/src/smtpd/smtpd.c
--- snapshot-20011127/src/smtpd/smtpd.c	Thu Nov 29 12:38:01 2001
+++ snapshot-20011127.dzikus/src/smtpd/smtpd.c	Thu Nov 29 12:32:18 2001
@@ -1899,6 +1899,7 @@
 	VAR_SMTPD_SND_AUTH_MAPS, DEF_SMTPD_SND_AUTH_MAPS, &var_smtpd_snd_auth_maps, 0, 0,
 	VAR_SMTPD_NOOP_CMDS, DEF_SMTPD_NOOP_CMDS, &var_smtpd_noop_cmds, 0, 0,
 	VAR_RELAY_CCERTS, DEF_RELAY_CCERTS, &var_relay_ccerts, 0, 0,
+	VAR_REJECT_REPLY_MSG_ACCESS_DENIED, DEF_REJECT_REPLY_MSG_ACCESS_DENIED, &var_reject_reply_msg_access_denied, 1, 0, 
 	0,
     };
 
diff -urN snapshot-20011127/src/smtpd/smtpd_check.c snapshot-20011127.dzikus/src/smtpd/smtpd_check.c
--- snapshot-20011127/src/smtpd/smtpd_check.c	Thu Nov 29 12:38:01 2001
+++ snapshot-20011127.dzikus/src/smtpd/smtpd_check.c	Thu Nov 29 12:32:18 2001
@@ -986,6 +986,7 @@
 			               char *reply_name, char *reply_class)
 {
     char   *myname = "check_relay_domains";
+	VSTRING *reply_msg = vstring_alloc(1);
 
     if (msg_verbose)
 	msg_info("%s: %s", myname, recipient);
@@ -1005,9 +1006,10 @@
     /*
      * Deny relaying between sites that both are not in relay_domains.
      */
+	vstring_strcat(reply_msg, "%d <%s>: %s ");
+	vstring_strcat(reply_msg, var_reject_reply_msg_access_denied);
     return (smtpd_check_reject(state, MAIL_ERROR_POLICY,
-			       "%d <%s>: %s odrzucony/rejected: Relay access denied",
-			       var_relay_code, reply_name, reply_class));
+			       vstring_export(reply_msg), var_relay_code, reply_name, reply_class));
 }
 
 /* permit_auth_destination - OK for message relaying */
@@ -1068,6 +1070,7 @@
 static int reject_unauth_destination(SMTPD_STATE *state, char *recipient)
 {
     char   *myname = "reject_unauth_destination";
+	VSTRING *reply_msg = vstring_alloc(1);
 
     if (msg_verbose)
 	msg_info("%s: %s", myname, recipient);
@@ -1081,9 +1084,10 @@
     /*
      * Reject relaying to sites that are not listed in relay_domains.
      */
+	vstring_strcat(reply_msg, "%d <%s>: ");
+	vstring_strcat(reply_msg, var_reject_reply_msg_access_denied);
     return (smtpd_check_reject(state, MAIL_ERROR_POLICY,
-			       "%d <%s>: Relay access denied",
-			       var_relay_code, recipient));
+			       vstring_export(reply_msg), var_relay_code, recipient));
 }
 
 /* reject_unauth_pipelining - reject improper use of SMTP command pipelining */
