diff -urN snapshot-19990518.orig/conf/aliases snapshot-19990518/conf/aliases
--- snapshot-19990518.orig/conf/aliases	Mon May 24 07:07:03 1999
+++ snapshot-19990518/conf/aliases	Mon May 24 07:10:13 1999
@@ -5,26 +5,29 @@
 #
 
 # Basic system aliases -- these MUST be present
+
+# Main system alias
+hostmaster: root
+
 MAILER-DAEMON: postmaster
-postmaster: root
+postmaster: hostmaster
 
 # General redirections for pseudo accounts
-bin:    root
-daemon: root
-named:  root
-nobody: root
-operator: root
-uucp:   root
-www:    root
-ftp-bugs: root
+bin:    hostmaster
+daemon: hostmaster
+named:  hosmaster
+nobody: hosmaster
+operator: hosmaster
+uucp:   hostmaster
+www:    hosmaster
+ftp-bugs: hostmaster
+ircmaster: hostmaster
+newsmaster: hostmaster
+webmaster: hostmaster
+sysadm: hostmaster
 
 # Put your local aliases here.
 
-# Well-known aliases -- these should be filled in!
-# root:
-# manager:
-# dumper:
-# operator:
 
 #++
 # NAME
diff -urN snapshot-19990518.orig/conf/main.cf snapshot-19990518/conf/main.cf
--- snapshot-19990518.orig/conf/main.cf	Mon May 24 07:07:03 1999
+++ snapshot-19990518/conf/main.cf	Mon May 24 07:17:14 1999
@@ -18,7 +18,7 @@
 # Postfix support programs and daemons. This setting can be overruled
 # with the command_directory and daemon_directory parameters.
 #
-program_directory = /some/where/postfix/bin
+program_directory = /usr/sbin
 
 # The command_directory parameter specifies the location of all
 # postXXX commands.  The default value is $program_directory.
@@ -40,14 +40,14 @@
 # OWNS NO OTHER FILES OR PROCESSES ON THE SYSTEM.  In particular,
 # don't specify nobody or daemon. PLEASE USE A DEDICATED USER.
 #
-#mail_owner = postfix
+mail_owner = postfix
 
 # The default_privs parameter specifies the default rights used by
 # the local delivery agent for delivery to external file or command.
 # These rights are used in the absence of a recipient user context.
 # DO NOT SPECIFY A PRIVILEGED USER OR THE POSTFIX OWNER.
 #
-#default_privs = nobody
+default_privs = nobody
 
 # INTERNET HOST AND DOMAIN NAMES
 # 
@@ -75,8 +75,8 @@
 # a domain-wide alias database that aliases each user to
 # user@that.users.mailhost.
 #
-#myorigin = $myhostname
-#myorigin = $mydomain
+myorigin = $myhostname
+myorigin = $mydomain
 
 # RECEIVING MAIL
 
@@ -85,7 +85,7 @@
 # the software claims all active interfaces on the machine. The
 # parameter also controls delivery of mail to user@[ip.address].
 #
-#inet_interfaces = all
+inet_interfaces = all
 #inet_interfaces = $myhostname
 #inet_interfaces = $myhostname, localhost
 
@@ -110,8 +110,8 @@
 #
 #mydestination = $myhostname, localhost.$mydomain
 #mydestination = $myhostname, localhost.$mydomain $mydomain
-#mydestination = $myhostname, localhost.$mydomain, $mydomain,
-#	mail.$mydomain, www.$mydomain, ftp.$mydomain
+mydestination = $myhostname, localhost.$mydomain, $mydomain,
+	mail.$mydomain, www.$mydomain, ftp.$mydomain
 
 # INTERNET VERSUS INTRANET
 #
@@ -179,7 +179,7 @@
 # "postfix reload" to eliminate the delay.
 #
 #alias_maps = dbm:/etc/aliases
-#alias_maps = hash:/etc/aliases
+alias_maps = hash:/etc/mail/aliases
 #alias_maps = hash:/etc/aliases, nis:mail.aliases
 
 # The alias_database parameter specifies the alias database(s) that
@@ -217,7 +217,7 @@
 # UNIX-style mailboxes are kept. The default setting depends on the
 # system type.
 #
-# mail_spool_directory = /var/mail
+mail_spool_directory = /var/mail
 # mail_spool_directory = /var/spool/mail
 
 # The mailbox_command parameter specifies the optional external
diff -urN snapshot-19990518.orig/conf/sample-aliases.cf snapshot-19990518/conf/sample-aliases.cf
--- snapshot-19990518.orig/conf/sample-aliases.cf	Mon May 24 07:07:03 1999
+++ snapshot-19990518/conf/sample-aliases.cf	Mon May 24 07:16:53 1999
@@ -12,7 +12,7 @@
 #
 # alias_database = dbm:/etc/aliases
 # alias_database = dbm:/etc/mail/aliases
-alias_database = hash:/etc/aliases
+alias_database = hash:/etc/mail/aliases
 
 # The alias_maps parameter specifies the list of alias databases used
 # by the local delivery agent. The default list is system dependent.
@@ -29,4 +29,4 @@
 # 
 # alias_maps = dbm:/etc/aliases, nis:mail.aliases
 # alias_maps = hash:/etc/aliases
-alias_maps = hash:/etc/aliases
+alias_maps = hash:/etc/mail/aliases
diff -urN snapshot-19990518.orig/global/mail_params.h snapshot-19990518/global/mail_params.h
--- snapshot-19990518.orig/global/mail_params.h	Mon May 24 07:07:02 1999
+++ snapshot-19990518/global/mail_params.h	Mon May 24 07:19:08 1999
@@ -180,7 +180,7 @@
   */
 #define VAR_CONFIG_DIR		"config_directory"
 #ifndef DEF_CONFIG_DIR
-#define DEF_CONFIG_DIR		"/etc/postfix"
+#define DEF_CONFIG_DIR		"/etc/mail"
 #endif
 extern char *var_config_dir;
 
diff -urN snapshot-19990518.orig/util/sys_defs.h snapshot-19990518/util/sys_defs.h
--- snapshot-19990518.orig/util/sys_defs.h	Mon May 24 07:07:02 1999
+++ snapshot-19990518/util/sys_defs.h	Mon May 24 07:19:31 1999
@@ -31,7 +31,7 @@
 #define HAS_DB
 #define HAS_SA_LEN
 #define DEF_DB_TYPE	"hash"
-#define ALIAS_DB_MAP	"hash:/etc/aliases"
+#define ALIAS_DB_MAP	"hash:/etc/mail/aliases"
 #define GETTIMEOFDAY(t)	gettimeofday(t,(struct timezone *) 0)
 #define ROOT_PATH	"/bin:/usr/bin:/sbin:/usr/sbin"
 #define USE_STATFS
