diff -uNr old-postfix-2.0.16/conf/main.cf postfix-2.0.16/conf/main.cf
--- old-postfix-2.0.16/conf/main.cf	Sat Sep  6 00:30:35 2003
+++ postfix-2.0.16/conf/main.cf	Thu Sep 25 17:28:00 2003
@@ -37,7 +37,7 @@
 # daemon programs (i.e. programs listed in the master.cf file). This
 # directory must be owned by root.
 #
-daemon_directory = /usr/libexec/postfix
+daemon_directory = /usr/lib/postfix
 
 # QUEUE AND PROCESS OWNERSHIP
 #
@@ -55,7 +55,7 @@
 # These rights are used in the absence of a recipient user context.
 # DO NOT SPECIFY A PRIVILEGED USER OR THE POSTFIX OWNER.
 #
-#default_privs = nobody
+default_privs = nobody
 
 # INTERNET HOST AND DOMAIN NAMES
 # 
@@ -87,7 +87,7 @@
 # myorigin also specifies the default domain name that is appended
 # to recipient addresses that have no @domain part.
 #
-#myorigin = $myhostname
+myorigin = $myhostname
 #myorigin = $mydomain
 
 # RECEIVING MAIL
@@ -377,7 +377,7 @@
 # "postfix reload" to eliminate the delay.
 #
 #alias_maps = dbm:/etc/aliases
-#alias_maps = hash:/etc/aliases
+alias_maps = hash:/etc/aliases
 #alias_maps = hash:/etc/aliases, nis:mail.aliases
 #alias_maps = netinfo:/aliases
 
@@ -388,7 +388,7 @@
 #
 #alias_database = dbm:/etc/aliases
 #alias_database = dbm:/etc/mail/aliases
-#alias_database = hash:/etc/aliases
+alias_database = hash:/etc/aliases
 #alias_database = hash:/etc/aliases, hash:/opt/majordomo/aliases
 
 # ADDRESS EXTENSIONS (e.g., user+foo)
@@ -416,7 +416,7 @@
 # UNIX-style mailboxes are kept. The default setting depends on the
 # system type.
 #
-#mail_spool_directory = /var/mail
+mail_spool_directory = /var/mail
 #mail_spool_directory = /var/spool/mail
 
 # The mailbox_command parameter specifies the optional external
@@ -438,8 +438,8 @@
 # IF YOU USE THIS TO DELIVER MAIL SYSTEM-WIDE, YOU MUST SET UP AN
 # ALIAS THAT FORWARDS MAIL FOR ROOT TO A REAL USER.
 #
-#mailbox_command = /some/where/procmail
-#mailbox_command = /some/where/procmail -a "$EXTENSION"
+#mailbox_command = /usr/bin/procmail
+#mailbox_command = /usr/bin/procmail -a "$EXTENSION"
 
 # The mailbox_transport specifies the optional transport in master.cf
 # to use after processing aliases and .forward files. This parameter
@@ -602,35 +602,10 @@
 #
 # The following parameters are used when installing a new Postfix version.
 # 
-# sendmail_path: The full pathname of the Postfix sendmail command.
-# This is the Sendmail-compatible mail posting interface.
-# 
-sendmail_path =
-
-# newaliases_path: The full pathname of the Postfix newaliases command.
-# This is the Sendmail-compatible command to build alias databases.
-#
-newaliases_path =
-
-# mailq_path: The full pathname of the Postfix mailq command.  This
-# is the Sendmail-compatible mail queue listing command.
-# 
-mailq_path =
+biff = no
 
 # setgid_group: The group for mail submission and queue management
 # commands.  This must be a group name with a numerical group ID that
 # is not shared with other accounts, not even with the Postfix account.
 #
-setgid_group =
-
-# manpage_directory: The location of the Postfix on-line manual pages.
-#
-manpage_directory =
-
-# sample_directory: The location of the Postfix sample configuration files.
-#
-sample_directory =
-
-# readme_directory: The location of the Postfix README files.
-#
-readme_directory =
+setgid_group = maildrop
diff -uNr old-postfix-2.0.16/src/global/mail_params.h postfix-2.0.16/src/global/mail_params.h
--- old-postfix-2.0.16/src/global/mail_params.h	Thu Sep 25 17:25:19 2003
+++ postfix-2.0.16/src/global/mail_params.h	Thu Sep 25 17:28:27 2003
@@ -64,7 +64,7 @@
 extern gid_t var_owner_gid;
 
 #define VAR_SGID_GROUP		"setgid_group"
-#define DEF_SGID_GROUP		"postdrop"
+#define DEF_SGID_GROUP		"maildrop"
 extern char *var_sgid_group;
 extern gid_t var_sgid_gid;
 
@@ -221,7 +221,7 @@
   */
 #define VAR_CONFIG_DIR		"config_directory"
 #ifndef DEF_CONFIG_DIR
-#define DEF_CONFIG_DIR		"/etc/postfix"
+#define DEF_CONFIG_DIR		"/etc/main"
 #endif
 extern char *var_config_dir;
 
diff -uNr old-postfix-2.0.16/src/util/sys_defs.h postfix-2.0.16/src/util/sys_defs.h
--- old-postfix-2.0.16/src/util/sys_defs.h	Thu Sep 11 00:33:31 2003
+++ postfix-2.0.16/src/util/sys_defs.h	Thu Sep 25 17:29:30 2003
@@ -129,7 +129,7 @@
 /* might be set by makedef */
 #ifdef HAS_DB
 #define DEF_DB_TYPE	"hash"
-#define ALIAS_DB_MAP	"hash:/etc/aliases"
+#define ALIAS_DB_MAP	"hash:/etc/mail/aliases"
 #else
 #define HAS_DBM
 #define	DEF_DB_TYPE	"dbm"
