diff -Nur snapshot-20011114.orig/src/global/mail_params.h snapshot-20011114/src/global/mail_params.h
--- snapshot-20011114.orig/src/global/mail_params.h	Thu Nov 15 15:13:42 2001
+++ snapshot-20011114/src/global/mail_params.h	Thu Nov 15 15:15:21 2001
@@ -961,6 +961,10 @@
 #define DEF_SMTPD_SASL_REALM	"$myhostname"
 extern char *var_smtpd_sasl_realm;
 
+#define VAR_PUBLISH_SASL_USERNAME	"publish_sasl_username"
+#define DEF_PUBLISH_SASL_USERNAME	0
+extern bool var_publish_sasl_username;
+
  /*
   * SASL authentication support, SMTP client side.
   */
Binary files snapshot-20011114.orig/src/smtpd/.smtpd.c.rej.swp and snapshot-20011114/src/smtpd/.smtpd.c.rej.swp differ
diff -Nur snapshot-20011114.orig/src/smtpd/smtpd.c snapshot-20011114/src/smtpd/smtpd.c
--- snapshot-20011114.orig/src/smtpd/smtpd.c	Thu Nov 15 15:13:42 2001
+++ snapshot-20011114/src/smtpd/smtpd.c	Thu Nov 15 15:20:19 2001
@@ -359,6 +359,7 @@
 bool    var_smtpd_sasl_enable;
 char   *var_smtpd_sasl_opts;
 char   *var_smtpd_sasl_realm;
+bool	var_publish_sasl_username;
 char   *var_filter_xport;
 bool    var_broken_auth_clients;
 char   *var_perm_mx_networks;
@@ -1011,6 +1012,13 @@
 		"Received: from %s (%s [%s])",
 		state->helo_name ? state->helo_name : state->name,
 		state->name, state->addr);
+#ifdef USE_SASL_AUTH
+	if (var_publish_sasl_username)
+	 rec_fprintf(state->cleanup, REC_TYPE_NORM,
+	  "\tsender authenticated as: \"%s\"",
+	  state->sasl_username ? state->sasl_username : "<none>");
+#endif
+	    
     if (var_smtpd_tls_received_header && state->tls_active) {
 	rec_fprintf(state->cleanup, REC_TYPE_NORM,
 		    "\t(using %s with cipher %s (%d/%d bits))",
@@ -1839,6 +1847,7 @@
 	VAR_DISABLE_VRFY_CMD, DEF_DISABLE_VRFY_CMD, &var_disable_vrfy_cmd,
 	VAR_ALLOW_UNTRUST_ROUTE, DEF_ALLOW_UNTRUST_ROUTE, &var_allow_untrust_route,
 	VAR_SMTPD_SASL_ENABLE, DEF_SMTPD_SASL_ENABLE, &var_smtpd_sasl_enable,
+	VAR_PUBLISH_SASL_USERNAME, DEF_PUBLISH_SASL_USERNAME, &var_publish_sasl_username,
 	VAR_BROKEN_AUTH_CLNTS, DEF_BROKEN_AUTH_CLNTS, &var_broken_auth_clients,
 	VAR_SMTPD_TLS_WRAPPER, DEF_SMTPD_TLS_WRAPPER, &var_smtpd_tls_wrappermode,
 	VAR_SMTPD_USE_TLS, DEF_SMTPD_USE_TLS, &var_smtpd_use_tls,
