--- snapshot-20010329/src/global/mail_params.h.wiget	Thu Apr 26 14:55:19 2001
+++ snapshot-20010329/src/global/mail_params.h	Thu Apr 26 14:55:20 2001
@@ -1423,6 +1423,13 @@
 #define DEF_SYSLOG_NAME			"postfix"
 extern char *var_syslog_name;
 
+ /*
+  * SMTPD messages
+  */
+#define VAR_REJECT_REPLY_MSG_ACCESS_DENIED	"smtpd_reject_msg_access_denied"
+#define DEF_REJECT_REPLY_MSG_ACCESS_DENIED	"rejected: Relay access denied."
+extern char *var_reject_reply_msg_access_denied;
+
 /* LICENSE
 /* .ad
 /* .fi
--- snapshot-20010329/src/global/mail_params.c.wiget	Thu Apr 26 14:55:19 2001
+++ snapshot-20010329/src/global/mail_params.c	Thu Apr 26 14:55:20 2001
@@ -65,6 +65,7 @@
 /*
 /*	char	*var_import_environ;
 /*	char	*var_export_environ;
+/*  char 	*var_reject_reply_msg_access_denied;
 /*
 /*	void	mail_params_init()
 /* DESCRIPTION
@@ -206,6 +207,7 @@
 
 char   *var_import_environ;
 char   *var_export_environ;
+char   *var_reject_reply_msg_access_denied;
 
 /* check_myhostname - lookup hostname and validate */
 
@@ -343,6 +345,7 @@
 	VAR_SMTP_TLS_CLIST, DEF_SMTP_TLS_CLIST, &var_smtp_tls_cipherlist, 0, 0,
 	VAR_SMTP_TLS_SCACHE_DB, DEF_SMTP_TLS_SCACHE_DB, &var_smtp_tls_scache_db, 0, 0,
 	VAR_TLS_DAEMON_RAND_SOURCE, DEF_TLS_DAEMON_RAND_SOURCE, &var_tls_daemon_rand_source, 0, 0,
+	VAR_REJECT_REPLY_MSG_ACCESS_DENIED, DEF_REJECT_REPLY_MSG_ACCESS_DENIED, &var_reject_reply_msg_access_denied, 1, 0, 
 	0,
     };
     static CONFIG_STR_FN_TABLE function_str_defaults_2[] = {
--- snapshot-20010329/src/smtpd/smtpd_check.c.wiget	Thu Apr 26 14:55:20 2001
+++ snapshot-20010329/src/smtpd/smtpd_check.c	Thu Apr 26 15:17:10 2001
@@ -823,6 +823,7 @@
 			               char *reply_name, char *reply_class)
 {
     char   *myname = "check_relay_domains";
+	VSTRING *reply_msg = vstring_alloc(1);
 
     if (msg_verbose)
 	msg_info("%s: %s", myname, recipient);
@@ -842,9 +843,10 @@
     /*
      * Deny relaying between sites that both are not in relay_domains.
      */
+	vstring_strcat(reply_msg, "%d <%s>: %s ");
+	vstring_strcat(reply_msg, var_reject_reply_msg_access_denied);
     return (smtpd_check_reject(state, MAIL_ERROR_POLICY,
-			       "%d <%s>: %s odrzucony/rejected: Relay access denied",
-			       var_relay_code, reply_name, reply_class));
+			       vstring_export(reply_msg), var_relay_code, reply_name, reply_class));
 }
 
 /* permit_auth_destination - OK for message relaying */
@@ -902,6 +904,7 @@
 static int reject_unauth_destination(SMTPD_STATE *state, char *recipient)
 {
     char   *myname = "reject_unauth_destination";
+	VSTRING *reply_msg = vstring_alloc(1);
 
     if (msg_verbose)
 	msg_info("%s: %s", myname, recipient);
@@ -915,9 +918,10 @@
     /*
      * Reject relaying to sites that are not listed in relay_domains.
      */
+	vstring_strcat(reply_msg, "%d <%s>: ");
+	vstring_strcat(reply_msg, var_reject_reply_msg_access_denied);
     return (smtpd_check_reject(state, MAIL_ERROR_POLICY,
-			       "%d <%s>: Relay access denied",
-			       var_relay_code, recipient));
+			       vstring_export(reply_msg), var_relay_code, recipient));
 }
 
 /* reject_unauth_pipelining - reject improper use of SMTP command pipelining */
--- snapshot-20010329/src/smtpd/smtpd.c.wiget	Thu Apr 26 14:55:19 2001
+++ snapshot-20010329/src/smtpd/smtpd.c	Thu Apr 26 14:55:20 2001
@@ -1734,6 +1734,7 @@
 	VAR_SMTPD_SASL_REALM, DEF_SMTPD_SASL_REALM, &var_smtpd_sasl_realm, 1, 0,
 	VAR_FILTER_XPORT, DEF_FILTER_XPORT, &var_filter_xport, 0, 0,
 	VAR_RELAY_CCERTS, DEF_RELAY_CCERTS, &var_relay_ccerts, 0, 0,
+	VAR_REJECT_REPLY_MSG_ACCESS_DENIED, DEF_REJECT_REPLY_MSG_ACCESS_DENIED, &var_reject_reply_msg_access_denied, 1, 0, 
 	0,
     };
 
