--- postfix-3.11.5.orig/src/global/dynamicmaps.c	2026-07-27 11:59:09.402204704 +0200
+++ postfix-3.11.5/src/global/dynamicmaps.c	2026-07-27 12:01:08.630019129 +0200
@@ -149,7 +149,7 @@ static const DICT_OPEN_INFO *dymap_dict_
      */
     if ((dp = (DYMAP_INFO *) htable_find(dymap_info, dict_type)) == 0) {
 	msg_warn("unsupported dictionary type: %s. "
-		 "Is the postfix-%s package installed?",
+		 "Is the postfix-dict-%s package installed?",
 		 dict_type, dict_type);
 	return (0);
     }
