--- postfix-3.11.5.orig/conf/main.cf	2026-07-27 11:59:09.224019293 +0200
+++ postfix-3.11.5/conf/main.cf	2026-07-27 12:00:21.726549083 +0200
@@ -62,7 +62,7 @@ command_directory = /usr/sbin
 # daemon programs (i.e. programs listed in the master.cf file). This
 # directory must be owned by root.
 #
-daemon_directory = /usr/libexec/postfix
+daemon_directory = /usr/lib/postfix
 
 # The data_directory parameter specifies the location of Postfix-writable
 # data files (caches, random numbers). This directory must be owned
@@ -86,7 +86,7 @@ mail_owner = postfix
 # These rights are used in the absence of a recipient user context.
 # DO NOT SPECIFY A PRIVILEGED USER OR THE POSTFIX OWNER.
 #
-#default_privs = nobody
+default_privs = nobody
 
 # INTERNET HOST AND DOMAIN NAMES
 # 
@@ -118,7 +118,7 @@ mail_owner = postfix
 # myorigin also specifies the default domain name that is appended
 # to recipient addresses that have no @domain part.
 #
-#myorigin = $myhostname
+myorigin = $myhostname
 #myorigin = $mydomain
 
 # RECEIVING MAIL
@@ -407,7 +407,7 @@ unknown_local_recipient_reject_code = 55
 # "postfix reload" to eliminate the delay.
 #
 #alias_maps = dbm:/etc/aliases
-#alias_maps = hash:/etc/aliases
+alias_maps = hash:/etc/mail/aliases
 #alias_maps = hash:/etc/aliases, nis:mail.aliases
 #alias_maps = netinfo:/aliases
 
@@ -418,7 +418,7 @@ unknown_local_recipient_reject_code = 55
 #
 #alias_database = dbm:/etc/aliases
 #alias_database = dbm:/etc/mail/aliases
-#alias_database = hash:/etc/aliases
+alias_database = hash:/etc/mail/aliases
 #alias_database = hash:/etc/aliases, hash:/opt/majordomo/aliases
 
 # ADDRESS EXTENSIONS (e.g., user+foo)
@@ -446,7 +446,7 @@ unknown_local_recipient_reject_code = 55
 # UNIX-style mailboxes are kept. The default setting depends on the
 # system type.
 #
-#mail_spool_directory = /var/mail
+mail_spool_directory = /var/mail
 #mail_spool_directory = /var/spool/mail
 
 # The mailbox_command parameter specifies the optional external
@@ -468,8 +468,8 @@ unknown_local_recipient_reject_code = 55
 # IF YOU USE THIS TO DELIVER MAIL SYSTEM-WIDE, YOU MUST SET UP AN
 # ALIAS THAT FORWARDS MAIL FOR ROOT TO A REAL USER.
 #
-#mailbox_command = /some/where/procmail
-#mailbox_command = /some/where/procmail -a "$EXTENSION"
+#mailbox_command = /usr/bin/procmail
+#mailbox_command = /usr/bin/procmail -a "$EXTENSION"
 
 # The mailbox_transport specifies the optional transport in master.cf
 # to use after processing aliases and .forward files. This parameter
@@ -641,54 +641,24 @@ debugger_command =
 #	-dmS $process_name gdb $daemon_directory/$process_name
 #	$process_id & sleep 1
 
+biff = no
+
 # INSTALL-TIME CONFIGURATION INFORMATION
 #
 # The following parameters are used when installing a new Postfix version.
-# 
-# sendmail_path: The full pathname of the Postfix sendmail command.
-# This is the Sendmail-compatible mail posting interface.
-# 
-sendmail_path =
-
-# newaliases_path: The full pathname of the Postfix newaliases command.
-# This is the Sendmail-compatible command to build alias databases.
-#
-newaliases_path =
-
-# mailq_path: The full pathname of the Postfix mailq command.  This
-# is the Sendmail-compatible mail queue listing command.
-# 
-mailq_path =
 
 # setgid_group: The group for mail submission and queue management
 # commands.  This must be a group name with a numerical group ID that
 # is not shared with other accounts, not even with the Postfix account.
 #
-setgid_group =
-
-# html_directory: The location of the Postfix HTML documentation.
-#
-html_directory =
+setgid_group = maildrop
 
-# manpage_directory: The location of the Postfix on-line manual pages.
-#
-manpage_directory =
+inet_protocols = ipv4
 
-# sample_directory: The location of the Postfix sample configuration files.
-# This parameter is obsolete as of Postfix 2.1.
+# The unknown_address_reject_code parameter specifies the SMTP server
+# response when a client violates the reject_unknown_sender_domain
+# or reject_unknown_recipient_domain restrictions.
 #
-sample_directory =
-
-# readme_directory: The location of the Postfix README files.
+# Contrary to what Wietse thinks (450) this should be 550
 #
-readme_directory =
-
-# shlib_directory: The location of the Postfix *.so library files and
-# database plugins.
-shlib_directory =
-
-# meta_directory: The location of non-executable files that are shared
-# between multiple Postfix instances.
-meta_directory = /etc/mail
-
-inet_protocols = ipv4
+unknown_address_reject_code = 550
--- postfix-3.11.5.orig/src/global/mail_params.h	2026-07-27 11:59:09.412311163 +0200
+++ postfix-3.11.5/src/global/mail_params.h	2026-07-27 11:59:49.280874790 +0200
@@ -116,7 +116,7 @@ extern uid_t var_owner_uid;
 extern gid_t var_owner_gid;
 
 #define VAR_SGID_GROUP		"setgid_group"
-#define DEF_SGID_GROUP		"postdrop"
+#define DEF_SGID_GROUP		"maildrop"
 extern char *var_sgid_group;
 extern gid_t var_sgid_gid;
 
--- postfix-3.11.5.orig/src/util/sys_defs.h	2026-07-27 11:59:09.514459752 +0200
+++ postfix-3.11.5/src/util/sys_defs.h	2026-07-27 12:00:34.123215747 +0200
@@ -777,7 +777,7 @@ extern int initgroups(const char *, int)
 #define HAS_FSYNC
 #define HAS_DB
 #define NATIVE_DB_TYPE	"hash"
-#define ALIAS_DB_MAP	"$default_database_type:/etc/aliases"
+#define ALIAS_DB_MAP	"$default_database_type:/etc/mail/aliases"
 #ifndef NO_NIS
 #define HAS_NIS
 #endif
