--- snapshot-20000507/global/mail_params.h.wiget	Mon May  8 01:42:27 2000
+++ snapshot-20000507/global/mail_params.h	Tue May  9 16:42:16 2000
@@ -187,7 +187,7 @@
   */
 #define VAR_CONFIG_DIR		"config_directory"
 #ifndef DEF_CONFIG_DIR
-#define DEF_CONFIG_DIR		"/etc/postfix"
+#define DEF_CONFIG_DIR		"/etc/mail"
 #endif
 extern char *var_config_dir;
 
--- snapshot-20000507/util/sys_defs.h.wiget	Sun May  7 19:30:37 2000
+++ snapshot-20000507/util/sys_defs.h	Tue May  9 16:42:16 2000
@@ -358,7 +358,7 @@
 #define HAS_FSYNC
 #define HAS_DB
 #define DEF_DB_TYPE	"hash"
-#define ALIAS_DB_MAP	"hash:/etc/aliases"
+#define ALIAS_DB_MAP	"hash:/etc/mail/aliases"
 #define HAS_NIS
 #define GETTIMEOFDAY(t)	gettimeofday(t,(struct timezone *) 0)
 #define ROOT_PATH	"/bin:/usr/bin:/sbin:/usr/sbin"
--- snapshot-20000507/conf/main.cf.wiget	Fri Jan 28 15:44:47 2000
+++ snapshot-20000507/conf/main.cf	Tue May  9 16:42:16 2000
@@ -24,7 +24,7 @@
 # default value is $program_directory. This directory must be owned
 # by root.
 #
-daemon_directory = /usr/libexec/postfix
+daemon_directory = /usr/lib/postfix
 
 # QUEUE AND PROCESS OWNERSHIP
 #
@@ -41,7 +41,7 @@
 # These rights are used in the absence of a recipient user context.
 # DO NOT SPECIFY A PRIVILEGED USER OR THE POSTFIX OWNER.
 #
-#default_privs = nobody
+default_privs = nobody
 
 # INTERNET HOST AND DOMAIN NAMES
 # 
@@ -69,7 +69,7 @@
 # a domain-wide alias database that aliases each user to
 # user@that.users.mailhost.
 #
-#myorigin = $myhostname
+myorigin = $myhostname
 #myorigin = $mydomain
 
 # RECEIVING MAIL
@@ -168,7 +168,7 @@
 # "postfix reload" to eliminate the delay.
 #
 #alias_maps = dbm:/etc/aliases
-#alias_maps = hash:/etc/aliases
+alias_maps = hash:/etc/mail/aliases
 #alias_maps = hash:/etc/aliases, nis:mail.aliases
 #alias_maps = netinfo:/aliases
 
@@ -179,7 +179,7 @@
 #
 #alias_database = dbm:/etc/aliases
 #alias_database = dbm:/etc/mail/aliases
-#alias_database = hash:/etc/aliases
+alias_database = hash:/etc/mail/aliases
 #alias_database = hash:/etc/aliases, hash:/opt/majordomo/aliases
 
 # ADDRESS EXTENSIONS (e.g., user+foo)
@@ -207,7 +207,7 @@
 # UNIX-style mailboxes are kept. The default setting depends on the
 # system type.
 #
-# mail_spool_directory = /var/mail
+mail_spool_directory = /var/mail
 # mail_spool_directory = /var/spool/mail
 
 # The mailbox_command parameter specifies the optional external
@@ -229,8 +229,8 @@
 # IF YOU USE THIS TO DELIVER MAIL SYSTEM-WIDE, YOU MUST SET UP AN
 # ALIAS THAT FORWARDS MAIL FOR ROOT TO A REAL USER.
 #
-#mailbox_command = /some/where/procmail
-#mailbox_command = /some/where/procmail -a "$EXTENSION"
+#mailbox_command = /usr/bin/procmail
+mailbox_command = /usr/bin/procmail -a "$EXTENSION"
 
 # The mailbox_transport specifies the optional transport in master.cf
 # to use after processing aliases and .forward files. This parameter
@@ -379,3 +379,4 @@
 	 xxgdb $daemon_directory/$process_name $process_id & sleep 5
 
 # Other configurable parameters.
+biff = no
