--- postfix-2.1.5/src/util/inet_addr_local.c.orig	2005-03-01 14:36:17.000000000 +0100
+++ postfix-2.1.5/src/util/inet_addr_local.c	2005-03-01 15:04:06.000000000 +0100
@@ -565,10 +565,10 @@
     if (addr_family == AF_INET6 || addr_family == AF_UNSPEC) {
 	count = addr_list->used;
 #ifdef INET6
-#if defined(HAS_PROCNET_IFINET6)
-	ial_procnet_ifinet6(addr_list, mask_list);
-#elif defined(HAVE_GETIFADDRS)
+#if defined(HAVE_GETIFADDRS)
 	ial_getifaddrs(addr_list, mask_list, AF_INET6);
+#elif defined(HAS_PROCNET_IFINET6)
+	ial_procnet_ifinet6(addr_list, mask_list);
 #else
 	ial_generic(addr_list, mask_list, AF_INET6);
 #endif
--- postfix-2.1.5/makedefs.orig	2005-03-01 14:36:17.000000000 +0100
+++ postfix-2.1.5/makedefs	2005-03-01 15:05:38.000000000 +0100
@@ -316,18 +316,6 @@
 # Check for IPv6 support
 
 if [ -z "$NO_IPV6" ] ; then
-if [ -f /usr/include/netinet6/in6.h ] ; then
-	grep __KAME__ /usr/include/netinet6/in6.h 2>&1 >/dev/null
-	if [ $?  = 1 ]; then
-		INET6=
-	else
-		if [ -f /usr/local/v6/lib/libinet6.a ]; then
-			INET6=kame
-		else
-			INET6=kame-merged
-		fi
-	fi
-fi
 if [ -z "$INET6" -a -f /usr/include/netinet/ip6.h ]; then
 	case "$SYSTYPE" in
 	SUNOS5)	INET6=solaris ;;
@@ -376,11 +364,6 @@
 	;;
 linux)
 	CCARGS="$CCARGS -DINET6 -D__ss_family=ss_family"
-	if test -f /usr/include/libinet6/netinet/ip6.h -a \
-		-f /usr/lib/libinet6.a; then 
-		CCARGS="$CCARGS -I/usr/include/libinet6 -DUSAGI_LIBINET6"
-		SYSLIBS="$SYSLIBS -linet6"
-	fi
 	;;
 esac
 
