diff -dur -x '*~' -x '*.orig' -x '*.rej' postfix-2.11.0.orig/conf/main.cf postfix-2.11.0/conf/main.cf
--- postfix-2.11.0.orig/conf/main.cf	2014-03-05 17:54:24.197406701 +0100
+++ postfix-2.11.0/conf/main.cf	2014-03-05 17:54:54.260740308 +0100
@@ -39,7 +39,7 @@
 # daemon programs (i.e. programs listed in the master.cf file). This
 # directory must be owned by root.
 #
-daemon_directory = /usr/libexec/postfix
+daemon_directory = /usr/lib/postfix
 
 # The data_directory parameter specifies the location of Postfix-writable
 # data files (caches, random numbers). This directory must be owned
@@ -63,7 +63,7 @@
 # These rights are used in the absence of a recipient user context.
 # DO NOT SPECIFY A PRIVILEGED USER OR THE POSTFIX OWNER.
 #
-#default_privs = nobody
+default_privs = nobody
 
 # INTERNET HOST AND DOMAIN NAMES
 # 
@@ -95,7 +95,7 @@
 # myorigin also specifies the default domain name that is appended
 # to recipient addresses that have no @domain part.
 #
-#myorigin = $myhostname
+myorigin = $myhostname
 #myorigin = $mydomain
 
 # RECEIVING MAIL
@@ -379,7 +379,7 @@
 # "postfix reload" to eliminate the delay.
 #
 #alias_maps = dbm:/etc/aliases
-#alias_maps = hash:/etc/aliases
+alias_maps = hash:/etc/mail/aliases
 #alias_maps = hash:/etc/aliases, nis:mail.aliases
 #alias_maps = netinfo:/aliases
 
@@ -390,7 +390,7 @@
 #
 #alias_database = dbm:/etc/aliases
 #alias_database = dbm:/etc/mail/aliases
-#alias_database = hash:/etc/aliases
+alias_database = hash:/etc/mail/aliases
 #alias_database = hash:/etc/aliases, hash:/opt/majordomo/aliases
 
 # ADDRESS EXTENSIONS (e.g., user+foo)
@@ -418,7 +418,7 @@
 # UNIX-style mailboxes are kept. The default setting depends on the
 # system type.
 #
-#mail_spool_directory = /var/mail
+mail_spool_directory = /var/mail
 #mail_spool_directory = /var/spool/mail
 
 # The mailbox_command parameter specifies the optional external
@@ -440,8 +440,8 @@
 # IF YOU USE THIS TO DELIVER MAIL SYSTEM-WIDE, YOU MUST SET UP AN
 # ALIAS THAT FORWARDS MAIL FOR ROOT TO A REAL USER.
 #
-#mailbox_command = /some/where/procmail
-#mailbox_command = /some/where/procmail -a "$EXTENSION"
+#mailbox_command = /usr/bin/procmail
+#mailbox_command = /usr/bin/procmail -a "$EXTENSION"
 
 # The mailbox_transport specifies the optional transport in master.cf
 # to use after processing aliases and .forward files. This parameter
@@ -613,45 +613,24 @@
 #	-dmS $process_name gdb $daemon_directory/$process_name
 #	$process_id & sleep 1
 
+biff = no
+
 # INSTALL-TIME CONFIGURATION INFORMATION
 #
 # The following parameters are used when installing a new Postfix version.
-# 
-# sendmail_path: The full pathname of the Postfix sendmail command.
-# This is the Sendmail-compatible mail posting interface.
-# 
-sendmail_path =
-
-# newaliases_path: The full pathname of the Postfix newaliases command.
-# This is the Sendmail-compatible command to build alias databases.
-#
-newaliases_path =
-
-# mailq_path: The full pathname of the Postfix mailq command.  This
-# is the Sendmail-compatible mail queue listing command.
-# 
-mailq_path =
 
 # setgid_group: The group for mail submission and queue management
 # commands.  This must be a group name with a numerical group ID that
 # is not shared with other accounts, not even with the Postfix account.
 #
-setgid_group =
-
-# html_directory: The location of the Postfix HTML documentation.
-#
-html_directory =
+setgid_group = maildrop
 
-# manpage_directory: The location of the Postfix on-line manual pages.
-#
-manpage_directory =
+inet_protocols = ipv4
 
-# sample_directory: The location of the Postfix sample configuration files.
-# This parameter is obsolete as of Postfix 2.1.
+# The unknown_address_reject_code parameter specifies the SMTP server
+# response when a client violates the reject_unknown_sender_domain
+# or reject_unknown_recipient_domain restrictions.
 #
-sample_directory =
-
-# readme_directory: The location of the Postfix README files.
+# Contrary to what Wietse thinks (450) this should be 550
 #
-readme_directory =
-inet_protocols = ipv4
+unknown_address_reject_code = 550
diff -dur -x '*~' -x '*.orig' -x '*.rej' postfix-2.11.0.orig/src/global/mail_params.h postfix-2.11.0/src/global/mail_params.h
--- postfix-2.11.0.orig/src/global/mail_params.h	2014-03-05 17:54:24.730740039 +0100
+++ postfix-2.11.0/src/global/mail_params.h	2014-03-05 17:54:54.260740308 +0100
@@ -67,7 +67,7 @@
 extern gid_t var_owner_gid;
 
 #define VAR_SGID_GROUP		"setgid_group"
-#define DEF_SGID_GROUP		"postdrop"
+#define DEF_SGID_GROUP		"maildrop"
 extern char *var_sgid_group;
 extern gid_t var_sgid_gid;
 
diff -dur -x '*~' -x '*.orig' -x '*.rej' postfix-2.11.0.orig/src/util/sys_defs.h postfix-2.11.0/src/util/sys_defs.h
--- postfix-2.11.0.orig/src/util/sys_defs.h	2014-03-05 17:54:24.944073374 +0100
+++ postfix-2.11.0/src/util/sys_defs.h	2014-03-05 17:58:20.204075514 +0100
@@ -769,7 +769,7 @@
 #define HAS_FSYNC
 #define HAS_DB
 #define NATIVE_DB_TYPE	"hash"
-#define ALIAS_DB_MAP	DEF_DB_TYPE ":/etc/aliases"
+#define ALIAS_DB_MAP	DEF_DB_TYPE ":/etc/mail/aliases"
 #ifndef NO_NIS
 #define HAS_NIS
 #endif
