--- snapshot-20010525/src/global/mail_params.h.wiget	Tue May 29 16:39:02 2001
+++ snapshot-20010525/src/global/mail_params.h	Tue May 29 16:39:02 2001
@@ -1434,6 +1434,13 @@
 #define DEF_SYSLOG_NAME			"postfix"
 extern char *var_syslog_name;
 
+ /*
+  * SMTPD messages
+  */
+#define VAR_REJECT_REPLY_MSG_ACCESS_DENIED	"smtpd_reject_msg_access_denied"
+#define DEF_REJECT_REPLY_MSG_ACCESS_DENIED	"rejected: Relay access denied."
+extern char *var_reject_reply_msg_access_denied;
+
 /* LICENSE
 /* .ad
 /* .fi
--- snapshot-20010525/src/global/mail_params.c.wiget	Tue May 29 16:39:02 2001
+++ snapshot-20010525/src/global/mail_params.c	Tue May 29 16:41:04 2001
@@ -67,6 +67,7 @@
 /*	char	*var_export_environ;
 /*	char	*var_debug_peer_list;
 /*	int	var_debug_peer_level;
+/*  char 	*var_reject_reply_msg_access_denied;
 /*
 /*	void	mail_params_init()
 /* DESCRIPTION
@@ -210,6 +211,7 @@
 char   *var_export_environ;
 char   *var_debug_peer_list;
 int     var_debug_peer_level;
+char   *var_reject_reply_msg_access_denied;
 
 /* check_myhostname - lookup hostname and validate */
 
@@ -348,6 +350,7 @@
 	VAR_SMTP_TLS_CLIST, DEF_SMTP_TLS_CLIST, &var_smtp_tls_cipherlist, 0, 0,
 	VAR_SMTP_TLS_SCACHE_DB, DEF_SMTP_TLS_SCACHE_DB, &var_smtp_tls_scache_db, 0, 0,
 	VAR_TLS_DAEMON_RAND_SOURCE, DEF_TLS_DAEMON_RAND_SOURCE, &var_tls_daemon_rand_source, 0, 0,
+	VAR_REJECT_REPLY_MSG_ACCESS_DENIED, DEF_REJECT_REPLY_MSG_ACCESS_DENIED, &var_reject_reply_msg_access_denied, 1, 0, 
 	0,
     };
     static CONFIG_STR_FN_TABLE function_str_defaults_2[] = {
--- snapshot-20010525/src/smtpd/smtpd_check.c.wiget	Tue May 29 16:39:02 2001
+++ snapshot-20010525/src/smtpd/smtpd_check.c	Tue May 29 16:39:02 2001
@@ -890,6 +890,7 @@
 			               char *reply_name, char *reply_class)
 {
     char   *myname = "check_relay_domains";
+	VSTRING *reply_msg = vstring_alloc(1);
 
     if (msg_verbose)
 	msg_info("%s: %s", myname, recipient);
@@ -909,9 +910,10 @@
     /*
      * Deny relaying between sites that both are not in relay_domains.
      */
+	vstring_strcat(reply_msg, "%d <%s>: %s ");
+	vstring_strcat(reply_msg, var_reject_reply_msg_access_denied);
     return (smtpd_check_reject(state, MAIL_ERROR_POLICY,
-			       "%d <%s>: %s odrzucony/rejected: Relay access denied",
-			       var_relay_code, reply_name, reply_class));
+			       vstring_export(reply_msg), var_relay_code, reply_name, reply_class));
 }
 
 /* permit_auth_destination - OK for message relaying */
@@ -970,6 +972,7 @@
 static int reject_unauth_destination(SMTPD_STATE *state, char *recipient)
 {
     char   *myname = "reject_unauth_destination";
+	VSTRING *reply_msg = vstring_alloc(1);
 
     if (msg_verbose)
 	msg_info("%s: %s", myname, recipient);
@@ -983,9 +986,10 @@
     /*
      * Reject relaying to sites that are not listed in relay_domains.
      */
+	vstring_strcat(reply_msg, "%d <%s>: ");
+	vstring_strcat(reply_msg, var_reject_reply_msg_access_denied);
     return (smtpd_check_reject(state, MAIL_ERROR_POLICY,
-			       "%d <%s>: Relay access denied",
-			       var_relay_code, recipient));
+			       vstring_export(reply_msg), var_relay_code, recipient));
 }
 
 /* reject_unauth_pipelining - reject improper use of SMTP command pipelining */
--- snapshot-20010525/src/smtpd/smtpd.c.wiget	Tue May 29 16:39:02 2001
+++ snapshot-20010525/src/smtpd/smtpd.c	Tue May 29 16:39:02 2001
@@ -1804,6 +1804,7 @@
 	VAR_SMTPD_SASL_REALM, DEF_SMTPD_SASL_REALM, &var_smtpd_sasl_realm, 1, 0,
 	VAR_FILTER_XPORT, DEF_FILTER_XPORT, &var_filter_xport, 0, 0,
 	VAR_RELAY_CCERTS, DEF_RELAY_CCERTS, &var_relay_ccerts, 0, 0,
+	VAR_REJECT_REPLY_MSG_ACCESS_DENIED, DEF_REJECT_REPLY_MSG_ACCESS_DENIED, &var_reject_reply_msg_access_denied, 1, 0, 
 	0,
     };
 
