--- postfix-3.3.1/src/global/dynamicmaps.c~	2018-09-15 23:09:41.000000000 +0200
+++ postfix-3.3.1/src/global/dynamicmaps.c	2018-09-15 23:10:31.630800874 +0200
@@ -171,14 +171,14 @@ static MKMAP_OPEN_FN dymap_mkmap_lookup(
      */
     if ((dp = (DYMAP_INFO *) htable_find(dymap_info, dict_type)) == 0)
 	msg_fatal("unsupported dictionary type: %s. "
-		  "Is the postfix-%s package installed?",
+		  "Is the postfix-dict-%s package installed?",
 		  dict_type, dict_type);
     if (!dp->mkmap_name)
 	msg_fatal("unsupported dictionary type: %s does not support "
 		  "bulk-mode creation.", dict_type);
     if (stat(dp->soname, &st) < 0)
 	msg_fatal("unsupported dictionary type: %s (%s: %m). "
-		  "Is the postfix-%s package installed?",
+		  "Is the postfix-dict-%s package installed?",
 		  dict_type, dp->soname, dict_type);
     if (st.st_uid != 0 || (st.st_mode & (S_IWGRP | S_IWOTH)) != 0)
 	msg_fatal("unsupported dictionary type: %s "
