--- src/global/mail_params.h.orig	2004-06-22 14:32:58.615734426 +0200
+++ src/global/mail_params.h	2004-06-22 14:33:21.260418134 +0200
@@ -2204,6 +2204,13 @@
 extern char *var_anvil_service;
 #endif
 
+ /*
+  * SMTPD messages
+  */
+#define VAR_REJECT_REPLY_MSG_ACCESS_DENIED	"smtpd_reject_msg_access_denied"
+#define DEF_REJECT_REPLY_MSG_ACCESS_DENIED	"rejected: Relay access denied."
+extern char *var_reject_reply_msg_access_denied;
+
 /* LICENSE
 /* .ad
 /* .fi
--- src/smtpd/smtpd.c.orig	2004-06-22 14:32:57.338808646 +0200
+++ src/smtpd/smtpd.c	2004-06-22 14:41:40.372360162 +0200
@@ -4514,6 +4514,7 @@
 	VAR_MILT_DEF_ACTION, DEF_MILT_DEF_ACTION, &var_milt_def_action, 1, 0,
 	VAR_MILT_DAEMON_NAME, DEF_MILT_DAEMON_NAME, &var_milt_daemon_name, 1, 0,
 	VAR_MILT_V, DEF_MILT_V, &var_milt_v, 1, 0,
+	VAR_REJECT_REPLY_MSG_ACCESS_DENIED, DEF_REJECT_REPLY_MSG_ACCESS_DENIED, &var_reject_reply_msg_access_denied, 1, 0,
 	0,
     };
     static CONFIG_RAW_TABLE raw_table[] = {
--- src/global/mail_params.c	2006-06-15 02:50:02.000000000 +0200
+++ src/global/mail_params.c-new	2006-07-11 19:29:07.162325784 +0200
@@ -78,6 +78,7 @@
 /*	char	*var_debug_peer_list;
 /*	int	var_debug_peer_level;
 /*	int	var_in_flow_delay;
+/*	char	*var_reject_reply_msg_access_denied;
 /*	int	var_fault_inj_code;
 /*	char   *var_bounce_service;
 /*	char   *var_cleanup_service;
@@ -246,6 +247,7 @@
 char   *var_export_environ;
 char   *var_debug_peer_list;
 int     var_debug_peer_level;
+char	*var_reject_reply_msg_access_denied;
 int     var_fault_inj_code;
 char   *var_bounce_service;
 char   *var_cleanup_service;
@@ -488,6 +490,7 @@
 	VAR_FLUSH_SERVICE, DEF_FLUSH_SERVICE, &var_flush_service, 1, 0,
 	VAR_VERIFY_SERVICE, DEF_VERIFY_SERVICE, &var_verify_service, 1, 0,
 	VAR_TRACE_SERVICE, DEF_TRACE_SERVICE, &var_trace_service, 1, 0,
+	VAR_REJECT_REPLY_MSG_ACCESS_DENIED, DEF_REJECT_REPLY_MSG_ACCESS_DENIED, &var_reject_reply_msg_access_denied, 1, 0,
 	0,
     };
     static CONFIG_STR_FN_TABLE function_str_defaults_2[] = {
