diff -ur postfix-19990906-pl05/global/own_inet_addr.c postfix-19990906-pl05.old/global/own_inet_addr.c
--- postfix-19990906-pl05/global/own_inet_addr.c	Tue Oct 12 21:00:22 1999
+++ postfix-19990906-pl05.old/global/own_inet_addr.c	Tue Oct 12 18:58:55 1999
@@ -109,8 +109,8 @@
 
     for (i = 0; i < addr_list.used; i++)
 	if (addr->__ss_family == addr_list.addrs[i].__ss_family) {
-	    if (!bcmp(addr->__ss_pad1, addr_list.addrs[i].__ss_pad1,
-		      addr->__ss_len))
+	    if (!bcmp(addr->__ss_padding, addr_list.addrs[i].__ss_padding,
+		      SA_LEN((struct sockaddr *)&addr)))
 	        return (1);
 	}
     return (0);
diff -ur postfix-19990906-pl05/smtp/smtp_addr.c postfix-19990906-pl05.old/smtp/smtp_addr.c
--- postfix-19990906-pl05/smtp/smtp_addr.c	Tue Oct 12 21:00:23 1999
+++ postfix-19990906-pl05.old/smtp/smtp_addr.c	Tue Oct 12 19:10:21 1999
@@ -320,14 +320,14 @@
 	    switch(addr->type) {
 		case T_A:
 		    if (self->addrs[i].__ss_family==AF_INET) {
-			if (!bcmp(self->addrs[i].__ss_pad1, addr->data,
+			if (!bcmp(self->addrs[i].__ss_padding, addr->data,
 				  sizeof(struct in_addr)))
 			    return(addr);
 		    }
 		    break;
 		case T_AAAA:
 		    if (self->addrs[i].__ss_family==AF_INET6) {
-			if (!bcmp(self->addrs[i].__ss_pad1, addr->data,
+			if (!bcmp(self->addrs[i].__ss_padding, addr->data,
 				  sizeof(struct in6_addr)))
 			    return(addr);
 		    }
