--- postfix-1.1.2/conf/master.cf.old	2002-03-02 20:13:56.000000000 -0300
+++ postfix-1.1.2/conf/master.cf	2002-05-02 21:44:39.000000000 -0300
@@ -108,7 +108,7 @@
   flags=R user=cyrus argv=/cyrus/bin/deliver -e -m ${extension} ${user}
 # Cyrus 2.1.5 (Amos Gouaux)
 cyrus	  unix	-	n	n	-	-	pipe
-  user=cyrus argv=/cyrus/bin/deliver -e -r ${sender} -m ${extension} ${user}
+  user=cyrus argv=/usr/lib/cyrus/deliver -e -r ${sender} -m ${extension} ${user}
 uucp	  unix	-	n	n	-	-	pipe
   flags=Fqhu user=uucp argv=uux -r -n -z -a$sender - $nexthop!rmail ($recipient)
 ifmail    unix  -       n       n       -       -       pipe
