--- postfix-1.1.11/src/util/dict_db.c.org	Thu Oct 31 14:39:28 2002
+++ postfix-1.1.11/src/util/dict_db.c	Thu Oct 31 14:38:52 2002
@@ -520,7 +520,7 @@
 	msg_fatal("set DB cache size %d: %m", DICT_DB_CACHE_SIZE);
     if (type == DB_HASH && db->set_h_nelem(db, DICT_DB_NELM) != 0)
 	msg_fatal("set DB hash element count %d: %m", DICT_DB_NELM);
-    if ((errno = db->open(db, db_path, 0, type, db_flags, 0644)) != 0)
+    if ((errno = db->open(db, NULL, db_path, 0, type, db_flags, 0644)) != 0)
 	msg_fatal("open database %s: %m", db_path);
     if ((errno = db->fd(db, &dbfd)) != 0)
 	msg_fatal("get database file descriptor: %m");
