diff -dur postfix-2.1.3.orig/conf/master.cf postfix-2.1.3/conf/master.cf
--- postfix-2.1.3.orig/conf/master.cf	2004-06-22 14:59:49.976722670 +0200
+++ postfix-2.1.3/conf/master.cf	2004-06-22 15:03:29.362874731 +0200
@@ -116,11 +116,11 @@
 # The Cyrus deliver program has changed incompatibly, multiple times.
 #
 old-cyrus unix  -       n       n       -       -       pipe
-  flags=R user=cyrus argv=/cyrus/bin/deliver -e -m ${extension} ${user}
+  flags=R user=cyrus argv=/usr/lib/cyrus/deliver -e -m ${extension} ${user}
 # Cyrus 2.1.5 (Amos Gouaux)
 # Also specify in main.cf: cyrus_destination_recipient_limit=1
 cyrus     unix  -       n       n       -       -       pipe
-  user=cyrus argv=/cyrus/bin/deliver -e -r ${sender} -m ${extension} ${user}
+  user=cyrus argv=/usr/lib/cyrus/deliver -e -r ${sender} -m ${extension} ${user}
 uucp      unix  -       n       n       -       -       pipe
   flags=Fqhu user=uucp argv=uux -r -n -z -a$sender - $nexthop!rmail ($recipient)
 ifmail    unix  -       n       n       -       -       pipe
