--- postfix-3.11.5.orig/conf/master.cf	2026-07-27 11:59:09.224225573 +0200
+++ postfix-3.11.5/conf/master.cf	2026-07-27 12:00:54.785821712 +0200
@@ -116,14 +116,14 @@ postlog   unix-dgram n  -       n
 # Also specify in main.cf: cyrus_destination_recipient_limit=1
 #
 #cyrus     unix  -       n       n       -       -       pipe
-#  flags=DRX user=cyrus argv=/cyrus/bin/deliver -e -r ${sender} -m ${extension} ${user}
+#  flags=DRX user=cyrus argv=/usr/sbin/deliver -e -r ${sender} -m ${extension} ${user}
 #
 # ====================================================================
 #
 # Old example of delivery via Cyrus.
 #
 #old-cyrus unix  -       n       n       -       -       pipe
-#  flags=R user=cyrus argv=/cyrus/bin/deliver -e -m ${extension} ${user}
+#  flags=R user=cyrus argv=/usr/sbin/deliver -e -m ${extension} ${user}
 #
 # ====================================================================
 #
